lib/mining/template.js

```diff
@@ -253,7 +254,7 @@ class BlockTemplate {
 
     // Smaller nonce for good measure.
     const nonce = Buffer.allocUnsafe(4);
-    nonce.writeUInt32LE(Math.random() * 0x100000000, 0, true);
+    nonce.writeUInt32LE(random.randomInt(), 0, true);
```

Comment:
Tested by generating 1000 blocks and checking the coinbase input script with bash. You can see the 4 byte nonce after the block height and coinbase flag, no errors or obvious pattern.

`for ((i = 0; i < 1000; ++i)); do bcoin-cli --network=regtest block $i | jq .txs[0].inputs[0].script ; done`

```
"510e6d696e65642062792062636f696e044c1669a6080000000000000000"
"520e6d696e65642062792062636f696e045b148e82080000000000000000"
"530e6d696e65642062792062636f696e043bd70072080000000000000000"
"540e6d696e65642062792062636f696e0450ca57bc080000000000000000"
"550e6d696e65642062792062636f696e0408ff06c3080000000000000000"
"560e6d696e65642062792062636f696e04e5f24a00080000000000000000"
"570e6d696e65642062792062636f696e0406ec9fe6080000000000000000"
"580e6d696e65642062792062636f696e04e56b3b6f080000000000000000"
"590e6d696e65642062792062636f696e04bafa7fb8080000000000000000"
"5a0e6d696e65642062792062636f696e0439c4e147080000000000000000"
"5b0e6d696e65642062792062636f696e0446bbcf6a080000000000000000"
"5c0e6d696e65642062792062636f696e049b1b9555080000000000000000"
"5d0e6d696e65642062792062636f696e04a6176549080000000000000000"
"5e0e6d696e65642062792062636f696e049e47076f080000000000000000"
"5f0e6d696e65642062792062636f696e04d638edbd080000000000000000"
"600e6d696e65642062792062636f696e04224cf192080000000000000000"
"01110e6d696e65642062792062636f696e0495a7753c080000000000000000"
"01120e6d696e65642062792062636f696e0446682bb2080000000000000000"
"01130e6d696e65642062792062636f696e04ef0b04b9080000000000000000"
"01140e6d696e65642062792062636f696e048e16f5a1080000000000000000"
"01150e6d696e65642062792062636f696e04b4ea18b0080000000000000000"
"01160e6d696e65642062792062636f696e047b8e9b91080000000000000000"
"01170e6d696e65642062792062636f696e04dfb6e25b080000000000000000"
"01180e6d696e65642062792062636f696e049be54125080000000000000000"
"01190e6d696e65642062792062636f696e042732df84080000000000000000"
"011a0e6d696e65642062792062636f696e04f49c66e9080000000000000000"
"011b0e6d696e65642062792062636f696e049d7bb9ff080000000000000000"
"011c0e6d696e65642062792062636f696e04b751fe7e080000000000000000"
"011d0e6d696e65642062792062636f696e0420e1e401080000000000000000"
"011e0e6d696e65642062792062636f696e0466f8de0e080000000000000000"
"011f0e6d696e65642062792062636f696e047a660974080000000000000000"
"01200e6d696e65642062792062636f696e04e92b9e5c080000000000000000"
"01210e6d696e65642062792062636f696e0474cc0271080000000000000000"
"01220e6d696e65642062792062636f696e04408b6c48080000000000000000"
"01230e6d696e65642062792062636f696e0443d788d8080000000000000000"
"01240e6d696e65642062792062636f696e049fa9f6b5080000000000000000"
"01250e6d696e65642062792062636f696e0467e06925080000000000000000"
"01260e6d696e65642062792062636f696e04788e6803080000000000000000"
"01270e6d696e65642062792062636f696e045f81432c080000000000000000"
"01280e6d696e65642062792062636f696e04652df227080000000000000000"
"01290e6d696e65642062792062636f696e042b144dfc080000000000000000"
"012a0e6d696e65642062792062636f696e0457eea140080000000000000000"
"012b0e6d696e65642062792062636f696e047508c4a8080000000000000000"
"012c0e6d696e65642062792062636f696e0413503d3e080000000000000000"
"012d0e6d696e65642062792062636f696e04358fa7d4080000000000000000"
"012e0e6d696e65642062792062636f696e042149de1c080000000000000000"
"012f0e6d696e65642062792062636f696e0448d1a21d080000000000000000"
"01300e6d696e65642062792062636f696e0412b0e674080000000000000000"
"01310e6d696e65642062792062636f696e0450801e57080000000000000000"
"01320e6d696e65642062792062636f696e0406efbe49080000000000000000"
"01330e6d696e65642062792062636f696e0492a2720e080000000000000000"
"01340e6d696e65642062792062636f696e04330b0b61080000000000000000"
"01350e6d696e65642062792062636f696e0469df3569080000000000000000"
```